### PR TITLE
[shtools] New version 4.9.1

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Comment by eschnett** [comment]
By the way, I needed to add `julia_compat="1.6"` manually because BinaryBuilder complained about that being necessary for experimental packages. If experimental packages are now the default, should this setting also be the default?

File: S/SHTOOLS/build_tarballs.jl [hunk 1]
```diff
@@ -2,10 +2,10 @@ using BinaryBuilder
 
 # Collection of sources required to build SHTOOLS
 name = "SHTOOLS"
-version = v"4.8"
+version = v"4.9.1"
 sources = [
-    ArchiveSource("https://github.com/SHTOOLS/SHTOOLS/releases/download/v4.8/SHTOOLS-4.8.tar.gz",
-                  "c36fc86810017e544abbfb12f8ddf6f101a1ac8b89856a76d7d9801ffc8dac44"),
+    ArchiveSource("https://github.com/SHTOOLS/SHTOOLS/releases/download/v4.9.1/SHTOOLS-4.9.1.tar.gz",
```
Comment (jeremiahpslewis) [inline S/SHTOOLS/build_tarballs.jl:7]:
```suggestion
    ArchiveSource("https://github.com/SHTOOLS/SHTOOLS/releases/download/v$(version)/SHTOOLS-$(version).tar.gz",
```

File: S/SHTOOLS/build_tarballs.jl [hunk 2]
```diff
@@ -14,19 +14,28 @@ script = raw"""
...
 
 # Build and install static libraries
 make fortran -j${nproc} F95FLAGS="-fPIC -O3 -std=gnu"
 make fortran-mp -j${nproc} F95FLAGS="-fPIC -O3 -std=gnu"
 make install PREFIX=${prefix}
 
 # Create shared libraries
-gfortran -shared -o ${libdir}/libSHTOOLS.${dlext} -Wl,$(flagon --whole-archive) ${prefix}/lib/libSHTOOLS.a -Wl,$(flagon --no-whole-archive | cut -d' ' -f1) -lfftw3 -lopenblas -lm
-gfortran -fopenmp -shared -o ${libdir}/libSHTOOLS-mp.${dlext} -Wl,$(flagon --whole-archive) ${prefix}/lib/libSHTOOLS-mp.a -Wl,$(flagon --no-whole-archive | cut -d' ' -f1) -lfftw3 -lopenblas -lm
+# TODO: Do we need to add " | cut -d' ' -f1"?
```
Comment (giordano) [inline S/SHTOOLS/build_tarballs.jl:28]: I thought I fixed that in #2988

File: S/SHTOOLS/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply S/SHTOOLS/build_tarballs.jl:7]: They don't seem to be very consistent with version tagging, I wouldn't count on that to work in v4.10

**Review by giordano — COMMENTED** [review]
>  If experimental packages are now the default, should this setting also be the default?

I'm not inclined to make `julia_compat="1.6"` the default because changing the julia compatibility bound _**requires**_ changing the version number of the package.  If we switched to the new compat bound automatically, we'd wreck packages that don't change the version number.  Getting the error alerts us that we need to something: either change the compat bound or filter out the new platforms

File: S/SHTOOLS/build_tarballs.jl (hunk 2 above)
Comment (eschnett) [reply S/SHTOOLS/build_tarballs.jl:28]: My assumption was that, if the package builds for all architectures, then this is fixed.

Comment (eschnett) [reply S/SHTOOLS/build_tarballs.jl:28]: It worked.

**Pull request merged**